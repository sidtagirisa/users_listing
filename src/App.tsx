import React from 'react';
import './App.css';
import StateProvider from './context/StateProvider';
import initialState from './context/initialState';
import Landing from './layouts/Landing';

function App() {
  return (
    <StateProvider initialState={initialState}>
      <Landing>
        <div>Hello World</div>
      </Landing>
    </StateProvider>
  );
}

export default App;
