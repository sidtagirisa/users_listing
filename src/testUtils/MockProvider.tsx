import React from 'react';
import StateProvider from '../context/StateProvider/index';
import initialState from '../context/initialState';

export default function MockProvider(props: {
  additionalStateRequired?: object;
  children: JSX.Element;
}): JSX.Element {
  const { additionalStateRequired, children } = props;

  return (
    <StateProvider
      initialState={{
        ...initialState,
        ...additionalStateRequired,
      }}
    >
      {children}
    </StateProvider>
  );
}

MockProvider.defaultProps = {
  value: {
    state: {},
    dispatch: jest.fn(),
  },
};
