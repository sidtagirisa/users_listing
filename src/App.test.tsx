import React from 'react';
import { render, act } from '@testing-library/react';
import App from './App';
import request from 'axios';
import { users } from './testUtils/fixtures';

describe('# App', () => {
  let requestMock: any;
  beforeEach(() => {
    requestMock = jest.spyOn(request, 'get');
  });

  it('should render basic app', async () => {
    requestMock.mockResolvedValue({ data: users });
    let container: any;

    await act(async () => {
      container = render(<App />).container;
    });

    expect(container.firstChild).toMatchSnapshot();
  });
});
