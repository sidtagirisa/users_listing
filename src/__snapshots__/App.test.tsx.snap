// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# App should render basic app 1`] = `
<div>
  Hello World
</div>
`;
